### ISSUE TITLE
`mad_ctpsa_hypot(3)` unexpected behavior

### ISSUE THREAD

**Issue by mattsignorelli** [MethodicalAcceleratorDesign/MAD-NG#417]
For complex scalars as TPSAs, `mad_ctpsa_hypot` gives:
```
hypot(3+3i, 4+4i) = 5+5i
hypot(3+3i, 4) = 4.4767839548935457 +2.0103717513913431i
``` 
In Julia/MATLAB/etc,  `hypot` with complex numbers always return a real scalar as `sqrt(cabs(a)^2+cabs(b)^2)`, so I'm not sure if this is a bug, or if not, then what exactly does `mad_ctpsa_hypot` calculate?

I also found `mad_tpsa_hypot` and `mad_ctpsa_hypot` do not correctly calculate hypot when using one of the input tpsas as the destination tpsa (e.g. `mad_tpsa_hypot(t1, t2, t1)` does not give the correct answer), and wanted to confirm this was expected behavior for this function, because all of the other ones so far appear ok with this.

**Comment by ldeniau** [comment]
tpsa and ctpsa functions do not perform type promotion (except those in mad_ctpsa_conv), i.e. they do not change the type returned with respect to the input type. Such promotions are managed at the LUA level.

I will check for the aliasing problem.

**Comment by ldeniau** [comment]
aliasing fixed in c785bf0f
